sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleMinMaxSumCount.java

```diff
@@ -0,0 +1,144 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import io.opentelemetry.sdk.metrics.data.MetricData.SummaryPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.ValueAtPercentile;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public final class DoubleMinMaxSumCount extends AbstractAggregator {
+
+  private static final AggregatorFactory AGGREGATOR_FACTORY =
+      new AggregatorFactory() {
+        @Override
+        public Aggregator getAggregator() {
+          return new DoubleMinMaxSumCount();
+        }
+      };
+
+  // The current value. This controls its own internal thread-safety via method access. Don't
+  // try to use its fields directly.
+  private final DoubleSummary current = new DoubleSummary();
+
+  public static AggregatorFactory getFactory() {
+    return AGGREGATOR_FACTORY;
+  }
+
+  private DoubleMinMaxSumCount() {}
+
+  @Override
+  void doMergeAndReset(Aggregator target) {
+    DoubleMinMaxSumCount other = (DoubleMinMaxSumCount) target;
+    current.mergeAndReset(other.current);
+  }
+
+  @Nullable
+  @Override
+  public Point toPoint(long startEpochNanos, long epochNanos, Map<String, String> labels) {
+    return current.toPoint(startEpochNanos, epochNanos, labels);
+  }
+
+  @Override
+  public void recordDouble(double value) {
+    current.record(value);
+  }
+
+  private static final class DoubleSummary {
+
+    private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+
+    @GuardedBy("lock")
+    private double sum = 0;
+
+    @GuardedBy("lock")
+    private long count = 0;
+
+    @GuardedBy("lock")
+    private double min = Double.POSITIVE_INFINITY;
+
+    @GuardedBy("lock")
+    private double max = Double.NEGATIVE_INFINITY;
+
+    private void record(double value) {
+      lock.writeLock().lock();
+      try {
+        count++;
+        sum += value;
+        min = Math.min(value, min);
+        max = Math.max(value, max);
+      } finally {
+        lock.writeLock().unlock();
+      }
+    }
+
+    public void mergeAndReset(DoubleSummary other) {
+      long myCount;
+      double mySum;
+      double myMin;
+      double myMax;
+      lock.writeLock().lock();
+      try {
+        myCount = this.count;
+        mySum = this.sum;
+        myMin = this.min;
+        myMax = this.max;
+        this.count = 0;
+        this.sum = 0;
+        this.min = Long.MAX_VALUE;
+        this.max = Long.MIN_VALUE;
+      } finally {
+        lock.writeLock().unlock();
+      }
```

Comment:
if this.count == 0 you can return immediately.

Reply:
true true

Reply:
also, the default values are wrong here.  fixing.